core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala

```diff
@@ -3782,6 +3784,33 @@ class GroupCoordinatorTest {
     assertTrue(groupCoordinator.tryCompleteHeartbeat(group, leaderMemberId, false, () => true))
   }
 
+  @Test
+  def testUpdateStaticMemberButStoreGroupFailed(): Unit = {
+    val rebalanceResult = staticMembersJoinAndRebalance(leaderInstanceId, followerInstanceId, DefaultSessionTimeout, DefaultRebalanceTimeout, "oldClientId", "oldClientHost")
+    val oldGroup = getGroup(groupId)
+    val oldMemberId = oldGroup.currentStaticMemberId(followerInstanceId)
+    val oldMember = oldGroup.get(oldMemberId.get)
+    val oldClientId = oldMember.clientId
+    val oldClientHost = oldMember.clientHost
+
+    val joinGroupResult = staticJoinGroupWithPersistence(groupId, JoinGroupRequest.UNKNOWN_MEMBER_ID,
+      followerInstanceId, protocolType, protocolSuperset, clockAdvance = 1, appendRecordError = Errors.MESSAGE_TOO_LARGE)
+
+    checkJoinGroupResult(joinGroupResult,
+      Errors.UNKNOWN_SERVER_ERROR,
+      rebalanceResult.generation,
+      Set.empty,
+      Stable,
+      Some(protocolType))
+
+    val newGroup = getGroup(groupId)
```

Comment:
nit: We don't need 2 separate variables for groupId. We can use the same `groupId` object.

Reply:
OK, let change it.